### ISSUE TITLE
Address Book for mobile

### ISSUE THREAD

**Issue by bdresser** [MetaMask/Design#126]
allow users to save addresses for simpler & safer sending

can be modeled closely after the extension's designs - are there any key platform differences that require modifications for mobile?  

dev issue on mobile board here: https://github.com/MetaMask/metamask-mobile/issues/128

**Comment by bdresser** [comment]
probably want some usability testing! @omnat

**Comment by omnat** [comment]
Sprint 21:
- [x] CJ having call with Esteban and Bruno to transfer extension designs to mobile

**Comment by cjeria** [comment]
Had a good call with the mobile devs today -  notes below. I also did some prior send flow pattern research, looking at how other apps implement contact list in send flows you can see [here](https://www.figma.com/file/DX3yjM7Rc2UzSRBgDL7RWN/Mobile-Contacts?node-id=10%3A32).

### Contact list In the send flow updates:
- Keep the network in top bar, as implemented (no switching ability)
- Display the to input/search bar below the top bar
- Show the **from** account in for context
- Add the ability to change the token (like a down arrow next to the currency symbol)
- Make **Amount** input more predominant - for easier interaction on mobile

Did not discuss the contact list for settings, left for the next phase.

**Comment by omnat** [comment]
One thing Rachel and I noticed today in the Send flow in extension.. perhaps good to re-consider for mobile too.

In the flow of 'Transfer between accounts', the chosen `from` account is also given as an option to select for `to` address. I believe there's no use-case when someone would want to send from Account 1 to Account 1, so we could help prevent an error here. 

So, in the list of accounts, lets show all of user accounts **except the `from` account**. wdyt @cjeria ?

**Comment by cjeria** [comment]
> In the flow of 'Transfer between accounts', the chosen from account is also given as an option to select for to address. I believe there's no use-case when someone would want to send from Account 1 to Account 1, so we could help prevent an error here.

Yes, we touched on this during design hand-off for the extension and I think it's a good constraint to add. I'll make a call out in the designs about it. We may also want to do that for the extension cc  @bdresser WDYT?

**Comment by omnat** [comment]
> In the flow of 'Transfer between accounts', the chosen from account is also given as an option to select for to address. I believe there's no use-case when someone would want to send from Account 1 to Account 1, so we could help prevent an error here.

Is this update of in dev backlog for extension? @bdresser 
We doing this in mobile?

For mobile, next steps (week of Sept 30):

- [ ] Add primary buttons to Send flow

- [ ] We'll show network on Amount tx screen, and which account you are sending the tx from

- [ ] Add token switcher within the send flow

A future thing to consider is a custom keyboard (issue created for this already in mobile dev backlog)

**Comment by benjaminaaron** [comment]
I love the new Contacts feature in the extension! It would be super useful to be able to sync it with mobile  💻🔄📱Is that planned by any chance?

As a DApp developer I would love to be able to have a way to use the contacts throughout Browser-DApps as well. Not just for sending transactions but as parameters for calling smart contract functions. Something like long/right-press on a text field and among copy/paste etc. there is a new option "insert address from contacts" 📖
Or having a way to access MetaMask contacts via the `web3` or unlocked `window.ethereum` object? So I could check if an inserted nickname has an entry in contacts 🔍 Maybe that is possible already?
(I posted the same on [a extension issue](https://github.com/MetaMask/metamask-extension/issues/7069#issuecomment-536712092))

**Comment by bdresser** [comment]
filed here @omnat @cjeria https://github.com/MetaMask/metamask-extension/issues/7237. i don't think this is super high priority, haven't heard any users having trouble with this, but should be simple to fix. 

@benjaminaaron yes, we do plan on syncing contacts - eventually! hang tight. no plans yet to make contacts available via the ethereum provider, there's a lot of user privacy issues to think through there, but i can understand how that'd be convenient for dapps.

**Comment by omnat** [comment]
Next steps in Sprint 23:
- Updates to prototype (per Figma notes)
- Milestone to hand-off to mobile for development

**Comment by cjeria** [comment]
Notes from design review:
- Updates to the design need to be made per notes in the [figma file](https://www.figma.com/file/DX3yjM7Rc2UzSRBgDL7RWN/Mobile-Contacts?node-id=309%3A1) 
- Feedback from Dan - display the "from" address in the send flow. How many users are mistakingly sending from the wrong account?

**Comment by omnat** [comment]
- @rachelcope to align 'Send/Submit' per the error designs updates

**Comment by omnat** [comment]
@cjeria to setup call with Esteban to transfer this

**Comment by cjeria** [comment]
We reviewed the latest implementation during the [mobile design QA call](https://consensys.slack.com/archives/GBW7S9FSN/p1576698012010500). 

Address book in send flow:

- [x] @christian.jeria to work on copy for error messages in the `to` screen. To reference copy we used in the extension
- [ ] Reconsider our current pattern `press/hold` to delete assets from asset list and contacts from inside settings - @estebanmino keep it press/hold. We'll internally test the build and see if interaction needs to be changed.
- [ ] Global primary currency settings - default to primary currency throughout send flow - @estebanmino to make this logic consistent
- [ ] in the `Amount` screen, if currency values are too long, they get chopped - @esteban.mino to look into using dynamically resize text here
- [ ] @estebanmino mino caught a bug on hex when sending ETH - the hex not showing in the small pop up
- [x] Internally test after building @estebanmino Usability concern when swapping Fiat to USD in the Amount screen. The native currency selector shows the token (e.g. ETH), not `USD`. We may want to set up a quick usability test for this.

Approve screen
- [ ] @cjeria to do design QA. After confirming an `Approve` call, we show a full-screen success confirmation and the toast. Let’s omit the full-screen success message and only show a toast message.

**Comment by cjeria** [comment]
Design feedback provided in this [doc](https://docs.google.com/document/d/1eB_Tq736L_r-w6kSkOFfomGrac7HopX5thFK51UvhBk/edit)
